src/main/java/com/gachigang/ontherun/config/handler/ApplicationExceptionHandler.java

```diff
@@ -1,88 +1,72 @@
+    private final ExceptionWebHandler exceptionWebHandler;
+    private final Environment environment;
 
     /**
-     * Handle all other exceptions.
+     * Handle AuthenticationExceptions.
      */
-    @ExceptionHandler(Exception.class)
-    public ResponseEntity<?> handleAllExceptions(Exception exception, HttpServletRequest request) {
-        log.error("Exception was thrown due unknown exception:", exception);
-
-        ResponseStatus responseStatus =
-                exception.getClass().getAnnotation(ResponseStatus.class);
-        HttpStatus status =
-                responseStatus != null ? responseStatus.value() : HttpStatus.INTERNAL_SERVER_ERROR;
-
-        ErrorDetails message = ErrorDetails.builder()
-                .status(status.value())
-                .timestamp(LocalDateTime.now())
-                .description(exceptionMessage.getUnauthorizedErrorMessage())
-                .url(request.getRequestURL().toString())
-                .build();
-        return new ResponseEntity<>(message, status);
+    @ExceptionHandler(JwtException.class)
+    public ResponseEntity<ErrorMessage> jwtException(@NonNull final AuthenticationException exception) {
+        log.error(exception.getLogMessage(), exception);
+        final String responseMessage = environment.getProperty(exception.getErrorMessageKey());
+        final HttpStatus responseHttpStatus = exceptionWebHandler.getHttpStatus(exception, HttpStatus.UNAUTHORIZED);
+        return exceptionWebHandler.getErrorResponse(exception, responseHttpStatus, responseMessage);
     }
 
     /**
      * Handle Not found exception.
      */
-    @ExceptionHandler(RuntimeException.class)
-    public ResponseEntity<ErrorDetails> handleNotFoundException(HttpServletRequest request,
-                                                                RuntimeException exception) {
-        ResponseStatus responseStatus =
-                exception.getClass().getAnnotation(ResponseStatus.class);
-        HttpStatus status =
-                responseStatus != null ? responseStatus.value() : HttpStatus.INTERNAL_SERVER_ERROR;
-        log.error("Exception was thrown because resource was not found:", exception);
-        ErrorDetails message = ErrorDetails.builder()
-                .status(status.value())
-                .timestamp(LocalDateTime.now())
-                .description(exceptionMessage.getNotFoundErrorMessage())
-                .url(request.getRequestURL().toString())
-                .build();
-        return ResponseEntity.status(status.value()).body(message);
+    @ExceptionHandler(ApplicationException.class)
+    public ResponseEntity<ErrorMessage> handleApplicationException(@NonNull final ApplicationException exception) {
+        log.error(exception.getLogMessage(), exception);
+        final String responseMessage = environment.getProperty(exception.getErrorMessageKey());
+        final HttpStatus responseHttpStatus = exceptionWebHandler.getHttpStatus(exception, HttpStatus.NOT_FOUND);
```

Comment:
Why do we have hardcoded HttpStatus.NOT_FOUND? As I remember we will return it if there are no HttpStatus annotation above exception -- It may provice some comfusing to us it future.

Reply:
Rewrited